Refined Git Backup System and Various Minor Edits

rajlego opened the pull request:
Made edits to content based on suggestions from people in SM discord. Added some optional suggestions for people to sync between computers and to ping commit.bat to taskbar. Integrated git setup with a MUCH easier method.

Please check: 

-tooltips/image formatting
-visual guide link works fine (is it fine to use HTML like that in MD? seems fine but just want to verify) 
-check the 'Optional: Sync SuperMemo between computers' section. It seems fine to me but I'm not sure if there's something dangerous in the script. I also don't know well enough to explain what the commands do, so if you can add a better explanation of them that would be good. 

Only meant to change: credits in root, qs-backup-setup.md, and added images to image folder in contents. Any changes outside of that should be reverted. Not sure why but sometimes my computer decides to mess up other files which has led me to errors like localhost:3000/#/ throwing 404 - Not Found for reasons I can't comprehend

rajlego committed:
Edits

Made edits to content based on suggestions from people in SM discord. Added some optional suggestions for people to sync between computers and to ping commit.bat to taskbar. Integrated git setup with a MUCH easier method.

alexis- committed:
- Added: .DS_Store and Thumbs.db files to the .gitignore
- Added: CREDITS.md which is the translation of credits.rtf into markdown
- Removed: .DS_Store and Thumbs.db files
- Removed: credits.rtf

alexis- commented on .gitignore:
```diff
@@ -1,2 +1,4 @@
 /docs/sma/content/videos/to-convert/ffmpeg2pass-0.log
 /docs/sma/content/videos/to-convert/ffmpeg2pass-0.log.mbtree
+.DS_STORE
+Thumbs.db
```
I have added these for future occasions

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -30,19 +30,24 @@ As of the moment of writing this guide (2020/02), this solution is **free** (as
 
 ###### Installing BitShelter
 
+?>Note that BitShelter will likely not run on 32 bit systems
```
- It is certain that BitShelter won't run on 32 bit systems at the moment, it is only compiled for x64
- There should be a space after `?>`. This isn't a big deal but having a consistent coding style helps a) keep the project clean and b) make it easier to quickly browse and edit the project

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -30,19 +30,24 @@ As of the moment of writing this guide (2020/02), this solution is **free** (as
+
 1. Download and install [the latest version](https://github.com/alexis-/BitShelter/releases) of *BitShelter*
-2. Start **BitShelter Agent** from the Windows Start menu
+2. Start **BitShelter Agent** from the Windows Start menu.
```
Not a big deal, but the other lines don't end with a dot

rajlego replied:
This is actually something I wanted to ask about, for the documentation should I add or not add periods after every line? Some places have them some places don't (though usually the sections are consistent)

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -51,13 +56,15 @@ You should obtain a layout similar to this one (it is fine if you only have one
-3. Select the SuperMemo drive **(E:\\)** in the [System Protection dialog ![](https://github.com/alexis-/BitShelter/raw/master/Resources/SystemPropertiesProtection_2018-05-05_13-10-18.png)](https://github.com/alexis-/BitShelter/raw/master/Resources/SystemPropertiesProtection_2018-05-05_13-10-18.png '@tooltip-preview') and click on <kbd>**Configure**</kbd>
-4. In the [new dialog ![](https://github.com/alexis-/BitShelter/raw/master/Resources/SystemPropertiesProtection_2018-05-05_13-10-22.png)](https://github.com/alexis-/BitShelter/raw/master/Resources/SystemPropertiesProtection_2018-05-05_13-10-22.png '@tooltip-preview') click on <kbd>**Turn on protection**</kbd>, select at least 2GB of desired reserved space(\*) for Snapshots, then press <kbd>**OK** ...
-5. Back in the [General tab ![](https://github.com/alexis-/BitShelter/raw/master/Resources/BitShelter.Agent_General.png)](https://github.com/alexis-/BitShelter/raw/master/Resources/BitShelter.Agent_General.png '@tooltip-preview'), click on <kbd>**Raise limit**</kbd>, and set the new limit to **512**
+1. In the [Main Window ![](https://raw.githubusercontent.com/alexis-/BitShelter/master/Resources/BitShelter.Agent_Rules.png)](https://raw.githubusercontent.com/alexis-/BitShelter/master/Resources/BitShelter.Agent_Rules.png '@tooltip-preview'), click on the <kbd>**Add Schedule**</kbd> button (If you get [this error ![](/content/images/backup-setup/bitshelter-error.png)](/content/images/backup-setu ...
```
This is important enough to assign it a distinct line and even a hint quote

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -51,13 +56,15 @@ You should obtain a layout similar to this one (it is fine if you only have one
+3. Select the SuperMemo drive **(E:\\)** in the [System Protection dialog ![](https://github.com/alexis-/BitShelter/raw/master/Resources/SystemPropertiesProtection_2018-05-05_13-10-18.png)](https://github.com/alexis-/BitShelter/raw/master/Resources/SystemPropertiesProtection_2018-05-05_13-10-18.png '@tooltip-preview') and click on <kbd>**Configure**</kbd>.
+4. In the [new dialog ![](https://github.com/alexis-/BitShelter/raw/master/Resources/SystemPropertiesProtection_2018-05-05_13-10-22.png)](https://github.com/alexis-/BitShelter/raw/master/Resources/SystemPropertiesProtection_2018-05-05_13-10-22.png '@tooltip-preview') click on <kbd>**Turn on protection**</kbd>, select at least 2GB of desired reserved space for Snapshots, then press <kbd>**OK**</kb ...
+
+!>If you chose not to create a dedicated partition for SuperMemo, increase reserve space significantly. As BitShelter will then run across your entire C:/ drive, it will be filled by all modifications made by any files in windows rather than SuperMemo alone.
```
It is best to reserve warning symbols for really important notes. This is more of an inconvenience. It could turn into an issue, but since we aren't explaining why it probably should be only a note.

For instance, this could be deserve a warning note:
```
 if you store your collection on C:\ with 5gb of snapshot space, if you download for 5gb or more of data (say, a HD movie), then you will effectively have flushed out all your previous snapshots
```

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
 
-Your BitShelter settings should look similar to the following configuration:
-<img src="/content/images/backup-setup/bitshelter-rules.png" data-origin="content/images/backup-setup/bitshelter-rules.png" alt="">
+Try out the following instructions and see if you're actually able to open up and use a copy made by BitShelter.
```
Excellent

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
 
+Your final BitShelter settings should look similar to the following configuration:
+
+<img src="/content/images/backup-setup/bitshelter-rules.png" data-origin="content/images/backup-setup/bitshelter-rules.png" alt="">
```
No idea why this was originally in html format. Better to use markdown where applicable

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
+
+1. Right click on your collection folder and click on **Restore previous versions**
+
+<img src="/content/images/backup-setup/bitshelter-restorepre.png" data-origin="content/images/backup-setup/bitshelter-restorepre.png" alt="">
```
Same

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
-<img src="/content/images/backup-setup/bitshelter-restore-previous-version.png" data-origin="content/images/backup-setup/bitshelter-restore-previous-version.png" alt="">
+2. Click on the copy you'd like to go back to:
+
+<img src="/content/images/backup-setup/bitshelter-restoremenu.png" data-origin="content/images/backup-setup/gitsetup-bitshelter-restoremenu" alt="">
```
Same

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
+
+3. You can then click restore, to overwrite current version of the folder, or click open to browse the files:
+
+<img src="/content/images/backup-setup/bitshelter-restoremenu2.png" data-origin="content/images/backup-setup/gitsetup-bitshelter-restoremenu2" alt="">
```
Same

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
 
-\*: GitHub has a 1GB limit on free private repositories.
-\*\*: *Although GitHub can be considered fairly reliable, we highly recommend that you implement a solution to encrypt your collection. This is, after all, a way to your most inner thoughts.*
+\*: GitHub has a 1GB limit on free private repositories. [GitLab](https://about.gitlab.com) offers 10gb free storage by default and [BitBucket](http://bitbucket.org) offers a 5gb plan for 3$. 
```
Best to try this out (GitLab) before adding it to the wiki.

alexis- replied:
I will leave this on the live version as the risk is minor, but make sure to check that this is a valid piece of advice (unless you already have, in that case disregard these two comments)

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
-2. [Follow this guide](https://vladmihalcea.com/tutorials/git/windows-git-ssh-authentication-to-github/) to:
-  - Install Git
-  - Setup a SSH key that automatically authenticate your Git installation with your GitHub account
+!> If you want to use SSH authentication instead of HTTPS, follow [this guide](https://vladmihalcea.com/tutorials/git/windows-git-ssh-authentication-to-github/)
```
Same. Warnings should really only be used to catch the user's attention on important information, otherwise it loses its meaning.

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
+1. Create a [new repository](https://github.com/new):
   - Give a name to your new repo (e.g. *SuperMemo-Collection*)
   - Select **Private**
+  - It is recommended to click **Initialize this repository with a README** to be able to immediately clone to computer
```
You should be able to clone the repository even without a README

alexis- replied:
I will remove this in the live version as I am confident about this information. Feel free to revert my change if I am mislead.

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
   - Select **Private**
+  - It is recommended to click **Initialize this repository with a README** to be able to immediately clone to computer
+
+<img src="/content/images/backup-setup/gitsetup-initialize.png" data-origin="content/images/backup-setup/gitsetup-initialize.png" alt="">
```
Same (markdown)

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
 
-4. In the **command prompt**, type `git clone <git@github.com:......>`. Replace the text between **< >** with the link you copied from GitHub.
-5. Your repository is now synchronized with your computer. Copy your SuperMemo collection in the new folder. Your folder should look similar to the example below (*.gitignore* might be missing):
+4. Copy the link starting with https://
```
Suggestion: Highlight the `https://` with back quotes. That's also consistent with the wiki style

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
-5. Your repository is now synchronized with your computer. Copy your SuperMemo collection in the new folder. Your folder should look similar to the example below (*.gitignore* might be missing):
+4. Copy the link starting with https://
+
+<img src="/content/images/backup-setup/gitsetup-clonehttps.png" data-origin="content/images/backup-setup/gitsetup-clonehttps.png" alt="">
```
Same (markdown)

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
+
+6. A popup will appear. Login with your GitHub account to authenticate.
+
+<img src="/content/images/backup-setup/gitsetup-login.png" data-origin="content/images/backup-setup/gitsetup-login.png" alt="">
```
Same (markdown)

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
+
+4.  Now, when you want to transfer from A to B or vice-versa, run `sm-main-commit.bat` on the computer you're transferring from and `sm-main-pull.bat` on the one you're transferring to
+
+!>Be extremely certain that you have no important files that haven't been uploaded before running sm-main-pull.bat as it will overrite all local changes
```
Space after `!> `

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
 
+
+
+#### Optional: Pin commit script to taskbar
```
Excellent

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
+
+To make running sm-main-commit.bat more convenient, you can add it to taskbar. Windows doesn't let you add .bat files to taskbar so instead you can do the following:
+
+1. Right click on sm-main-commit.bat and click [create shortcut ![](/content/images/backup-setup/shortcut-creation.png)](/content/images/backup-setup/shortcut-creation.png '@tooltip-preview').
```
FYI: The url here is a very special case, because it starts with a `/`. What this means is that it will look from the root domain (e.g. `https://supermemo.wiki/content/...` if your domain is `supermemo.wiki` and you are using https (as opposed to http)) instead of the relative, current folder (e.g. `https://supermemo.wiki/sma/content/...` if you're in the sma folder).

The reason for this is that we include the backup tutorial both at the root (SM wiki) and in /sma/ (SMA wiki), but we don't want to duplicate the files.

alexis- commented on docs/qs-backup-setup.md:
````diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
+
+1. Right click on sm-main-commit.bat and click [create shortcut ![](/content/images/backup-setup/shortcut-creation.png)](/content/images/backup-setup/shortcut-creation.png '@tooltip-preview').
+
+2. The new shortcut still can't be added to taskbar. As such, right click properties and [go to shortcuts menu ![](/content/images/backup-setup/shortcut-edit.png)](/content/images/backup-setup/shortcut-edit.png '@tooltip-preview'). Add ```cmd /c``` (```cmd \c``` will not work) to the front of [ the target script ![](/content/images/backup-setup/shortcut-target.png)](/content/images/backup-setup/s ...
````
It would be best if `/content/images/backup-setup/shortcut-target.png` showed E:\SuperMemo-Collection` for the drive instead of `C:\SuperMemo-Collection`, since this is the drive we have been illustrating this guide with.

alexis- replied:
Why end with semi colon `:` ?

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
+
+#### Optional: Sync SuperMemo between computers
+
+!>As a slight mishap could cause data loss, don't try this without understanding the git commands involved. To learn more about how git works, check out [this guide](https://hackernoon.com/understanding-git-fcffd87c15a3) or this [interactive tutorial](https://learngitbranching.js.org/)
```
Space after `!>`

alexis- commented on docs/qs-backup-setup.md:
```diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
+
+The starting computer will be referred to as 'A' and the target computer you want to sync your collection to will be 'B'
+
+1. On computer A,  <a href="/content/data/sm-main-pull.bat" target="_blank" rel="noopener">download this .bat file</a> and save it in your local repository folder (where your `.git` directory is located). It contains the following commands:
```
Same (image url markdown)

alexis- commented on docs/qs-backup-setup.md:
````diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
+git fetch --all
+git reset --hard origin/master
+```
+?>These commands override the local repository's files for the ones that you've most recently uploaded to git
````
Space after `?>`

alexis- commented on docs/qs-backup-setup.md:
````diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
+git fetch --all
+git reset --hard origin/master
+```
+?>These commands override the local repository's files for the ones that you've most recently uploaded to git
````
That's very dangerous. If you have uncommitted work, you will lose all of it after running `git reset --hard origin/master`. I'm not sure that we want to advertise such a method.

rajlego replied:
Is there a better way to write the script that would be safer? I was thinking that if there are local changes you could have it automatically commit to a backup branch and after that overwrite with origin/master

alexis- replied:
I think this calls for a user decision. You could add a prompt dialog:

In Batch: https://stackoverflow.com/questions/50763301/how-to-add-a-yes-no-response-in-a-batch-file
In Powershell: https://stackoverflow.com/questions/24649019/how-to-use-confirm-in-powershell/24649481

alexis- replied:
To be honest, I don't feel very comfortable with this. I am not saying we shouldn't add it to the wiki, but I can't seem to shake this feeling.

I think it is one thing to suggest that users can synchronize their files across computers. But it is another one to provide an official method.

With the former, the user is responsible for making sure that his method doesn't endanger his collection. With the latter, we can be blamed if something goes wrong. It might also lull users with a false sense of security. E.g. `If it's on supermemo.wiki, and boy are they picky with safety, then it must ok to do this.`

If we do this, it should ideally be carefully planned. We can provide some guidance, with a Caveat Emptor in big, red letters and warning signs on both sides, but I'm not a fan of it.

With of all that being said, and at the risk of contradicting myself, I still believe yours is the best choice. Users will try to synchronize their collection with or without our help anyway. So we might as well get them started, and avoid them some pitfalls if we can.

alexis- commented on docs/qs-backup-setup.md:
````diff
@@ -82,81 +89,155 @@ You should obtain a layout similar to this one (it is fine if you only have one
+
+2. The new shortcut still can't be added to taskbar. As such, right click properties and [go to shortcuts menu ![](/content/images/backup-setup/shortcut-edit.png)](/content/images/backup-setup/shortcut-edit.png '@tooltip-preview'). Add ```cmd /c``` (```cmd \c``` will not work) to the front of [ the target script ![](/content/images/backup-setup/shortcut-target.png)](/content/images/backup-setup/s ...
+
+3. You can now [right click the shortcut ![](/content/images/backup-setup/shortcut-taskbar.png)](/content/images/backup-setup/shortcut-taskbar.png '@tooltip-preview') and add it to taskbar. After you are done using SuperMemo, click the shortcut in taskbar (or use winkey + taskbar position number key) to sync changes to your collections's Github repository.
````
`or use winkey + taskbar position number key`

use &lt;kbd&gt; notation instead. e.g. 
![image](https://user-images.githubusercontent.com/3876562/77360580-e6c80900-6d4d-11ea-80b7-8a52e2faa80f.png)
(see SMA installation wiki page)

alexis- reviewed (changes requested):
- The naming of file is really good. Two minor improvements I would suggest:
  - Rename `gitsetup-xxx` files into `git-setup-xxx`. This makes it easier to read and will increase the chances to find these files by either searching for "git" or "setup" across different OS (not all OS handle search in the same way)
  - Rename `gitsetup-clonehttps.png` to `git-setup-clone-https.png` for the same reasons
- Your last commit added several `.DS_Store` and one `Thumbs.db` files. I have removed them
- I have added these files to the .gitignore (as the name implies, it will ignore ignore these files starting from your next commits)
- See https://github.com/zanfab/docsify-plugin-flexible-alerts for additional quote styles (note, hint, warning, etc.)
- Kudos on getting all the links right, and with tooltip

You have a knack for details, good work.

alexis- commented:
I will proceed to apply some of these changes right now as discussed

alexis- committed:
- Added: Backup-setup > Details for the Git setup (thanks @rajlego) #51
- Added: Backup-setup > Simpler setup for Git authentication (HTTPS with credential manager instead of SSH) (thanks @rajlego) #51
- Added: Flexible Alerts Docsify plugin (see index.html)
- Updated: Backup-setup > Several minor improvements (thanks @rajlego) #51
- Updated: Edited & cropped bitshelter-restorepre.png to reduce its height

alexis- commented:
PR merged, but some issues are still outstanding (see review)
